File: Data Structures & Algorithms/two-integer-sum/submission-5.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {
        const obj = new Map();


        for(let i = 0; i < nums.length; i++){
            let diff = target - nums[i];
            if(obj.has(diff)){
                return [i, obj.get(diff)]
            }

            obj.set(nums[i], i)
        }

        return []
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-5.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        i = 0
        y = len(numbers) - 1

        while i < y:
            sum = numbers[i] + numbers[y]
            if sum == target:
                return [i + 1, y + 1]
            elif sum < target:
                i += 1
            else:
                y -= 1
        
            

